storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -6952,6 +6953,8 @@ static void xtrabackup_prepare_func(int argc, char **argv) {
       }
 
       /* Write MySQL 8.0 .cfg file */
+      /* load the dynamic metadata */
+      dict_table_load_dynamic_metadata(table);
```

Comment:
if we load right value here, do we still need the above "autoinc_persisted != table->autoinc_persisted" change?

Reply:
yes, we would still need these changes.  autoinc != table->autoinc doesn't work when table has just one row. 

If table has just one row. then table->autoinc would be 1. Also, MySQL initialise autoinc with 1. So this condition will not be satisfied. 

I proposed the fix of using autoinc_persisted because server initialise autoinc_persisted with 0. and after loading the dynamic metadata table-> autoinc_persisted would be 1. 

if you want  limit the scope of changes by introducing another condition autoinc =1 ?